win_acl_inheritance

h0nIg opened the pull request:
added dedicated module to disable ACL inheritance. this could be merged into win_acl, but windows acl are ways more complex than linux permissions/ACL and therefore the module is already very complex (e.g. lot of parameters). this is just an addon to win_acl and focused on removing the inheritance.

future: maybe additional parameter could be added to reconvert ACL to enabled inheritance

robynbergeron commented:
Greetings, @h0nIg --

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes, which can be contributed by anyone other than the author of the new module:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

More information about the review process can be seen here: 
https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md

Thanks again for submitting your Ansible module!

robynbergeron commented:
Pinging @schwartzmx & @trondhindenes for feedback as the win_acl module owners, who may have feedback here --

And pinging the rest of our Windows crew for thoughts as well: @petemounce @elventear @smadam813 @jhawkesworth @angstwad @sivel @chrishoffman @cchurch

marcind commented on windows/win_acl_inheritance.py:
```diff
@@ -0,0 +1,59 @@
+# Playbook example
+---
+- name: Disable and copy
+  win_owner:
```
Wrong module name

marcind commented on windows/win_acl_inheritance.py:
```diff
@@ -0,0 +1,59 @@
+    copy: yes
+
+- name: Disable
+  win_owner:
```
Wrong module name

marcind commented on windows/win_acl_inheritance.py:
```diff
@@ -0,0 +1,59 @@
+version_added: "2.0"
+short_description: Disable ACL inheritance
+description:
+    - Disable ACL inheritance and optionally converts ACE to dedicated ACE
```
Would be helpful for Windows novices to expand the 3-letter acronyms on first mention. Eg. `Disable ACL (Access Control List) inheritance ...`

marcind commented on windows/win_acl_inheritance.ps1:
```diff
@@ -0,0 +1,55 @@
+Set-Attr $result "changed" $false;
+
+$path = Get-Attr $params "path" -failifempty $true
+$copy = Get-Attr $params "copy" "no" -validateSet "no","yes" -resultobj $result
```
Might want to use [`ConvertTo-Bool`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/powershell.ps1#L163) so that the value of `$copy` is a bool

h0nIg commented:
added suggested changes and modified module to enable/disable inheritance. in addition, added feature to remove ACE's that are already present as dedicated ACE's

gregdek commented:
Pinging the Windows crew for re-review: @trondhindenes @petemounce @elventear @smadam813 @jhawkesworth @angstwad @sivel @chrishoffman @cchurch

mojo4life commented:
works_for_me

celticur commented:
looks good, but please correct the version_added, since 2.0 has been released already.

passes_guidelines

h0nIg commented:
corrected

schwartzmx commented:
lgtm, although haven't tested
passes_guidelines

h0nIg commented:
would be nice if some of the core guys will comment on this @abadger @sivel @gregdek @bcoca

Jmainguy commented:
@h0nIg See if you can squash the commits if at all possible. Since this is extras you only need 1 passes_guidlines, and one works_for_me. So this should be ready for inclusion according to those rules. @robynbergeron ^^

h0nIg commented:
@Jmainguy checkout the "Files changed" tab at the top. there you can see the full changeset

nitzmahone commented on windows/win_acl_inheritance.ps1:
```diff
@@ -0,0 +1,85 @@
+
+$path = Get-Attr $params "path" -failifempty $true
+$state = Get-Attr $params "state" "absent" -validateSet "present","absent" -resultobj $result
+$reorganize = Get-Attr $params "reorganize" "no" -validateSet "no","yes" -resultobj $result | ConvertTo-Bool
```
The inline `| ConvertTo-Bool` bombs hard (swallowing the error message) if an invalid value is passed (eg, "true" instead of "yes"). Maybe drop it down to the next line?

nitzmahone commented:
@h0nIg This is great- a couple little things and we can get this one in. It feels like reorganize should maybe be two different args- maybe something like "copy_inherited_on_removal" (only valid for state absent) and "deduplicate_inherited_aces" (only valid with state present). Naming things is hard, but I'd probably prefer to err on the side of transparency about what it does. The reorganize behaviors could also probably use a little expansion in the docs on what exactly is going on- I can take a swing at that part if you'd rather.

nitzmahone commented:
@h0nIg also, you've been writing enough Windows modules that I imagine you'd have some thoughts on one of the things I have slated for 2.1: a new Windows module API that mirrors the python one. Let me know if you want to kick stuff around for that- I'd love your input.

gregdek commented:
@h0nIg A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

h0nIg commented:
added extensive documentation suggestion / get-attr + converttobool suggestion by @nitzmahone . this should be ready for inclusion :)

h0nIg commented:
ready_for_review

gregdek commented:
Thanks @h0nIg for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

h0nIg commented:
@gregdek this pull request already received two passes_guidlines and two works_for_me. just @nitzmahone asked for a minor improvement related to documentation / input variables.

could you please merge this, since people are not reacting and its hard to get "shipit" again (which was equal to passes_guidlines / works_for_me before "Ansibull-bot")

gregdek commented:
Based on comments, I'll put it into shipit and leave the decision to @nitzmahone and others.

h0nIg commented:
thanks @gregdek , could you please check https://github.com/ansible/ansible-modules-extras/pull/1117 too, since its the same situation (2 received,  @nitzmahone acknowledged too)

bcoca commented:
cc @nitzmahone

gregdek commented:
@nitzmahone please revisit. This one is close, we'd like to get it over the finish line. :)
